### PR TITLE
Move logic to improve performance.

### PR DESCRIPTION
**Pull request by spacedmonkey**
Trac ticket: https://core.trac.wordpress.org/ticket/58235

Follow on from https://core.trac.wordpress.org/changeset/56037. 

Move the check to see if properties exist before doing anymore processing. 

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/media.php [hunk 1]
```diff
@@ -1931,6 +1931,11 @@ function wp_filter_content_tags( $content, $context = null ) {
  * @return string Converted `img` tag with optimization attributes added.
  */
 function wp_img_tag_add_loading_optimization_attrs( $image, $context ) {
+	// Images should have source and dimension attributes for the loading optimization attributes to be added.
```
Comment (swissspidy) [inline src/wp-includes/media.php:1934]: The regexes below do check for single quotes and double quotes. Why don't these checks do that? Seems like an oversight.

(And yes I know this is just moved from elsewhere, but now is a great opportunity to discuss this and fix/document it)

File: src/wp-includes/media.php [hunk 2]
```diff
@@ -1931,6 +1931,11 @@ function wp_filter_content_tags( $content, $context = null ) {
...
+	if ( ! str_contains( $image, ' src="' ) || ! str_contains( $image, ' width="' ) || ! str_contains( $image, ' height="' ) ) {
+		return $image;
+	}
```
Comment (felixarntz) [inline src/wp-includes/media.php:1937]: This change must not be made. Please see the comment starting in line 1939/1944.

File: src/wp-includes/media.php [hunk 3]
```diff
@@ -2164,6 +2164,11 @@ function wp_iframe_tag_add_loading_attr( $iframe, $context ) {
 		return $iframe;
 	}
 
+	// Iframes should have source and dimension attributes for the `loading` attribute to be added.
+	if ( ! str_contains( $iframe, ' src="' ) || ! str_contains( $iframe, ' width="' ) || ! str_contains( $iframe, ' height="' ) ) {
+		return $iframe;
+	}
```
Comment (felixarntz) [inline src/wp-includes/media.php:2170]: Same here, this must not be moved here.

**Review by felixarntz — CHANGES_REQUESTED** [review]
@spacedmonkey Thanks for opening this PR for consideration. Looking at the change, the main change you made here is not something we can do, please review the code comment that explains why.

The only other change here is moving the `preg_match_all()` call in `wp_filter_content_tags()`, which makes sense to me. However, that change is unrelated to #58235, so I think that should be a separate ticket.

**Comment by spacedmonkey** [comment]
@felixarntz Good catch. Sorry I missed that comment how. Closing this PR.

**Pull request closed without merging**